### DISCUSSION TITLE
Is it possible to create dynamic forms?

### DISCUSSION THREAD

**Discussion by jmbds** [avo-hq/avo#899] [General]
Imagine we a have a class A and 2 classes B and C that inherit from class A.

`class A < ApplicationRecord`
`class B < A`
`class C < A`

I wanted to show every resource of class B and C in the same index view and when on create view show/hide fields according to the value of the field type (a select). Is this possible?

**Comment by adrianthedev** [comment]
Hey @jmbds.

That looks like [STI](https://guides.rubyonrails.org/association_basics.html#single-table-inheritance-sti). If that's the case, Avo supports [that](https://docs.avohq.io/2.0/associations.html#single-table-inheritance-sti).

Check out the implementation on [avodemo](https://avodemo.herokuapp.com/avo/resources/people/24/edit).

https://github.com/avo-hq/avodemo/blob/main/app/models/person.rb
https://github.com/avo-hq/avodemo/blob/main/app/avo/resources/person_resource.rb
https://github.com/avo-hq/avodemo/blob/main/app/models/spouse.rb
https://github.com/avo-hq/avodemo/blob/main/app/avo/resources/spouse_resource.rb
